### ISSUE TITLE
With your package Xcode downloads 361.20 MiB and it takes several minutes 🤯

### ISSUE THREAD

**Issue by TofPlay** [OneSignal/OneSignal-iOS-SDK#817]
**Description:**
Every time I download your SDK via Xcode it downloads 361.20 MiB and it takes several minutes 🤯

```
$ git clone https://github.com/OneSignal/OneSignal-iOS-SDK
Cloning into 'OneSignal-iOS-SDK'...
remote: Enumerating objects: 31, done.
remote: Counting objects: 100% (31/31), done.
remote: Compressing objects: 100% (24/24), done.
remote: Total 9893 (delta 12), reused 10 (delta 7), pack-reused 9862
Receiving objects: 100% (9893/9893), 361.20 MiB | 1.85 MiB/s, done.
Resolving deltas: 100% (7063/7063), done.
```

The problem is that you put your XCFramework in github when it is not necessary.
If you want to make a Package.swift with an XCFramework you generate a zip of the XCFramework that you put in a CDN in https (important the https!). Only the Package.swift should be put on github.
I invite you to take look at the script that I made [framework2swiftpm](https://github.com/TofPlay/framework2swiftpm) that will undoubtedly help you.

**Comment by rgomezp** [comment]
Howdy @TofPlay ,
Thanks for your suggestion. We will review it and make a decision.

**Comment by emawby** [comment]
This is fixed as of release 3.1.1, but it requires Xcode 12. If you are using Xcode 12 please use the OneSignal-XCFramework Swift Package instead of using the OneSignal-iOS-SDK.
